client/src/image-uploader.tsx

```diff
@@ -7,25 +7,31 @@ export const ImageUploader = () => {
     const { updateAppState } = useContext(AppContext);
     const theme = useTheme();
 
-    const onDrop = useCallback((acceptedFiles: File[]) => {
-        if(acceptedFiles.length === 0)
-            return;
-
+    const onDrop = useCallback(async (acceptedFiles: File[]) => {
+        if (acceptedFiles.length === 0) return;
+    
         const file = acceptedFiles[0];
-        const blobUrl = URL.createObjectURL(file);
-
+        const preview = URL.createObjectURL(file);
+    
+        updateAppState({ 
+            image: { file, preview } 
+        });
+    
         const image = new Image();
-        image.src = blobUrl;
-
+        image.src = preview;
+        
         image.onload = () => {
             const { naturalWidth: width, naturalHeight: height } = image;
-            const canvas = new OffscreenCanvas(width, height);
-            const ctx = canvas.getContext("2d");
+            const canvas = document.createElement('canvas');
+            canvas.width = width;
+            canvas.height = height;
```

Comment:
why did you change it? Offscreen canvas looks like a proper way to render image and not show it to the user?